rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt

```diff
@@ -1532,3 +1532,51 @@ data class XFootStmt(
         xfoot(this, interpreter)
     }
 }
+
+@Serializable
+data class SubstStmt(
+    val length: Expression?,
+    val value: Expression,
+    val startPosition: Expression?,
+    val target: AssignableExpression,
+    val operationExtender: String?,
+    override val position: Position? = null
+) : Statement(position) {
+    override fun execute(interpreter: InterpreterCore) {
+
+        val start = if (startPosition?.let { interpreter.eval(it).asString() } != null) {
```

Comment:
Maybe we can improve all tests regarding the "null safety" by using let scoped function and|or the kotlin Elvis operator﻿

Reply:
Changed code using Elvis operator